### PR TITLE
Update retrieving of Sonar test artefacts

### PR DESCRIPTION
**Pull request by chudilka1**

### CODE REVIEW

File: .github/workflows/ci-core.yml [hunk 1]
```diff
@@ -183,8 +183,8 @@ jobs:
         with:
           name: ${{ matrix.cmd }}_${{ matrix.split.idx }}_logs
           path: |
-            ./output.txt
-            ./race.*
+            ./test-report.json
+            ./race-coverage.txt
```
Comment (jmank88) [inline .github/workflows/ci-core.yml:187]: We still need to upload any detected races too.

File: tools/bin/go_core_race_tests [hunk 2]
```diff
@@ -4,7 +4,7 @@ set -ex
 TIMEOUT="${TIMEOUT:-30s}"
 COUNT="${COUNT:-10}"
 GO_LDFLAGS=$(bash tools/bin/ldflags)
-GORACE="log_path=$PWD/race" go test -tags test -race -ldflags "$GO_LDFLAGS" -shuffle on -timeout "$TIMEOUT" -count "$COUNT" $1 | tee ./output.txt
+GORACE="log_path=$PWD/race" go test -tags test -race -ldflags "$GO_LDFLAGS" -shuffle on -timeout "$TIMEOUT" -count "$COUNT" -covermode=atomic -coverpkg=./... -coverprofile=race-coverage.txt $1 | tee output.txt
```
Comment (jmank88) [inline tools/bin/go_core_race_tests:7]: Does the `./` on `output.txt` matter?

**Review by jmank88 — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Comment by jmank88** [comment]
This looks fine, although race coverage might be a little bit odd and unstable, since we do random shuffling and don't always complete all of the tests. Will the analysis still be useful if the results are inconsistent?

**Comment by chudilka1** [comment]
> This looks fine, although race coverage might be a little bit odd and unstable, since we do random shuffling and don't always complete all of the tests. Will the analysis still be useful if the results are inconsistent?

Yes, as we collect non-race unit tests coverage as well. So eventually, the coverage reports will be merged and provide cumulative metrics.

**Pull request closed without merging**